### DISCUSSION TITLE
What is the link path for alternate languages

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2341] [Q&A] [answered]
Since Material now has multi-language support since v7 am I in the process of translating my own documentation into another language.
A question I have however is, what the `link` option for each entry in the `alternate` option is. Is it a path relative to the `docs` directory? Relative to the root directory?
Also, how would this affect the navigation in the mkdocs folder?

**Comment by squidfunk** [comment, marked as answer]
The link should be an absolute path to the alternate language, the `site_url` of the alternate language if you will. Currently, switching the language on any site will always transport you to this exact link. In the future (not too far from now), it will be supported to link a specific page to another page, e.g.:

``` sh
https://example.com/en/*                -> https://example.com/de/              # Now
https://example.com/en/getting-started/ -> https://example.com/de/vorbereitung/ # Future
https://example.com/en/releases/        -> https://example.com/de/versionen/    # Future
```

Note that there is also a `lang` attribute, which is currently not used, but will be very helpful for SEO in the near future.

**Reply by Andre601** [reply]
This is rather odd tbh.
I used `docs/en` and it didn't work, but using just `en` worked. The folder structure was of course `docs/en`

I hope there will be better options in the future as the current way (and documentation about it) is a bit weird to understand and if you like me, want to have all langs inside the docs directory, will mkdocs obviously complain about certain files not being defined in the nav section...

**Reply by squidfunk** [reply]
There has to be a separate installation for each language, because otherwise, it wouldn't be possible to tell MkDocs to use a different language for every page (the `theme.lang` property), which is crucial to be able to use the correct stemmer for every language at least. Also, your German translation's TOC wouldn't be called "Inhaltsverzeichnis" but "Table of Contents".

#2346 is a detailed take on how to best pull of multi-language installations, with shared assets and so on. Note that you could also use the language selector to just sublink to other pages if you wish, as it's (currently) just links after all.

**Reply by Andre601** [reply]
Thanks for linking the guide. Will take a look.
